### DISCUSSION TITLE
Custom Trace ID and Span ID

### DISCUSSION THREAD

**Discussion by rdoshi29** [open-telemetry/opentelemetry-java#4028] [Q&A] [answered]
Is there any way to set custom Trace IDs and Span IDs to Open-telemetry traces in java?

**Comment by jkwatson** [comment, marked as answer]
If you implement your own IdGenerator, you can plug that into your SDK and use that. You'll need to conform to the otel traceid/spanid requirements with the ids you generate.

**Reply by gauravc95** [reply]
How do you plug customized IdGenerator into the SDK? Also, if you have any sample code can you share?

**Reply by jkwatson** [reply]
Start here: https://opentelemetry.io/docs/instrumentation/java/manual/#set-up The `SdkTracerProviderBuilder` has the option to set it.
